test/test_torch.py

```diff
@@ -9534,20 +9534,76 @@ def test_rpow(self, device):
         assert m.dim() == 0, "m is intentionally a scalar"
         self.assertEqual(torch.pow(2, m), 2**m)
 
+    @precisionOverride({torch.float32: 1e-5, torch.complex64: 1e-5})
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    @dtypes(torch.double)
+    @dtypesIfCPU(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
     def test_symeig(self, device, dtype):
-        from torch.testing._internal.common_utils import random_symmetric_matrix
+        from torch.testing._internal.common_utils import random_hermitian_matrix
 
         def run_test(dims, eigenvectors, upper):
```

Comment:
nit - let's move `run_test` outside `test_symeig` so that both `test_symeig` and `test_symeig_complex_failed` can just call into it.

Reply:
The `test_symeig_complex_xfailed` test shall be removed once batched matmul works with complex on GPU. I hope that it happens soon. So I'm thinking not to move `run_test` outside, but I'll make the test lighter than just copy-pasting.